dogstatsd/src/datadog.rs

```diff
@@ -193,6 +207,7 @@ impl DdApi {
         url: String,
         body: Vec<u8>,
         content_type: &str,
+        retry_strategy: RetryStrategy,
```

Comment:
`ship_data` has access to `self`, can't you just access to it by doing `self.retry_strategy.clone()` instead of passing it on every other method?

Reply:
done